torch/csrc/jit/passes/onnx/peephole.cpp

```diff
@@ -19,20 +19,34 @@ bool isBroadcasting(Node *node) {
   return broadcasting.count(node->kind());
 }
 
-// When iterating over the dimension sizes, starting at the trailing dimension,
-// the dimension sizes must either be equal, or one of them does not exist.
+// First iterate over the 'from' tensor sizes. Ignore all leading and trailing
+// dimensions that are simply one, since they can be trivially broadcasted.
+// When iterating over the dimension sizes (with reduced 'from' tensor),
+// starting at the trailing dimension, the dimension sizes must either be equal,
+// or one of them does not exist.
```

Comment:
It's worth calling out that these semantics are NOT numpy style broadcasting, and they are not the generalized broadcasting that PyTorch implements in general.